src/shared_modules/content_manager/src/components/offlineDownloader.hpp

```diff
@@ -80,49 +83,98 @@ class OfflineDownloader final : public AbstractHandler<std::shared_ptr<UpdaterCo
     };
 
     /**
-     * @brief Downloads the requested local file and update the context accordingly.
+     * @brief Copy a file from the localsystem.
      *
-     * @note Despite the method name, there is no such download since the file is present in the filesystem.
+     * @param inputFilepath Input path from where to copy the file.
+     * @param outputFilepath Output path where to paste the file.
+     * @return true if the file was copied, otherwise false.
+     */
+    bool copyFile(const std::filesystem::path& inputFilepath, const std::filesystem::path& outputFilepath) const
+    {
+        constexpr auto FILE_PREFIX {"file://"};
+
+        // Remove file prefix.
+        auto unprefixedUrl {inputFilepath.string()};
+        Utils::replaceFirst(unprefixedUrl, FILE_PREFIX, "");
+
+        // Check input file existence.
+        if (!std::filesystem::exists(unprefixedUrl))
+        {
+            std::cout << "File " << unprefixedUrl << " doesn't exist. Skipping download." << std::endl;
+            return false;
```

Comment:
Question: If the download from an HTTP server fails, then an exception is thrown why is this not the case for this scenario?

Reply:
Let's keep the same behavior in both cases.

Reply:
Done!